accepted/future-releases/inline-classes/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+Assume that _DV_ declares a final instance variable with name `id` and
+type `R`.
+
+We say that the _declared representation type_ of `Inline`
+is `R`, and the _instantiated representation type_ corresponding to
+<code>Inline&lt;T<sub>1</sub>,.. T<sub>s</sub>&gt;</code> is
+<code>[T<sub>1</sub>/X<sub>1</sub>, .. T<sub>s</sub>/X<sub>s</sub>]R</code>.
+
+We will omit 'declared' and 'instantiated' from the phrase when it is
+clear from the context whether we are talking about the inline class
+itself, or we're talking a particular instantiation of a generic
+inline. *For non-generic inline classes, the representation type is
+the same in either case.*
+
+Let `V` be an inline type of the form
+<code>Inline&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>,
+and let `R` be the corresponding instantiated representation type.
+`V` is a proper subtype of `Object?`. If `R` is non-nullable then `V`
+is a proper subtype of `Object` as well.
+
+*That is, an expression of an inline type can be assigned to a top type
+(like all other expressions), and if the representation type is
+non-nullable then it can also be assigned to `Object`. Non-inline types
+(except bottom types) cannot be assigned to inline types without a cast.*
+
+In the body of a member of an inline class declaration _DV_ named
+`Inline` and declaring the type parameters
+<code>X<sub>1</sub>, .. X<sub>s</sub></code>,
+the static type of `this` is
+<code>Inline&lt;X<sub>1</sub> .. X<sub>s</sub>&gt;</code>.
+The static type of the representation name is the representation
+type.
+
+*For example, in `inline class V { final R id; ...}`, `id` has type
+`R`, and `this` has type `V`.*
+
+Again, let `V` be an inline type of the form
+<code>Inline&lt;T<sub>1</sub>, .. T<sub>s</sub>&gt;</code>,
+and let `R` be the corresponding instantiated representation type.
+If `R` is not an inline type then we say that `V` is an inline type
+at level zero. If `R` is an inline type at level _k_ then we say that
+`V` is an inline type at level _k + 1_.
+A compile-time error occurs if the level of `V` is undefined.
+
+*In other words, cycles are not allowed.*
+
+An inline class declaration _DV_ named `Inline` may declare one or
+more constructors. A constructor which is declared in an inline class
+declaration is also known as an _inline class constructor_.
+
+*The purpose of having an inline class constructor is that it bundles
+an approach for building an instance of the representation type of an
+inline declaration _DV_ with _DV_ itself, which makes it easy to
+recognize that this is a way to obtain a value of that inline type. It
+can also be used to verify that an existing object (provided as an
+actual argument to the constructor) satisfies the requirements for
+having that inline type.*
+
+A compile-time error occurs if an inline class constructor includes a
+superinitializer. *That is, a term of the form `super(...)` or
+`super.id(...)` as the last element of the initializer list.*
```

Comment:
Could we allow it? If I *want* to run the constructor of the super-class? 
It must necessarily pass the representation object to the super-inline-class, which is hard to specify, so maybe it's better to just make you call the super-constructor in the body.

Reply:
Considering that we're going with `implements` rather than `extends` for composing inline classes, I think we are doing the right thing by not allowing `super(...)` or `super.id(...)`.

Reply:
I would recommend that we make it an error now. A future enhancement could include the ability to run these constructors using special syntax. It is of course already possible to simply invoke them (and, probably, ignore the value of the expression).

Reply:
Very good. I'll conclude that it is an error to have a superinitializer and resolve this thread.